lib/po_libc_wrappers.c

```diff
@@ -107,7 +108,11 @@ open(const char *path, int flags, ...)
 	mode = va_arg(args, int);
 	rel = find_relative(path, NULL);
 
-	return openat(rel.dirfd, rel.relative_path, flags, mode);
+	// If the file is already opened, no need of relative opening!
+	if( strcmp(rel.relative_path,".") != 0 )
+		return openat(rel.dirfd, rel.relative_path, flags, mode);
+	else
+		return rel.dirfd;
```

Comment:
Rather than returning the FD directly, we should return a `dup(2)`'d version. Code that calls `open(2)` expects to receive a file descriptor that's independent of what's going on in, e.g., other threads. If we just return the original stored descriptor, we could end up with multiple `open(2)` invocations that return the same descriptor. If that happened, a `read(2)` on that descriptor by one thread (for example) would affect the seek index experienced by another thread that thinks it has its own descriptor but, in reality, is using the same descriptor.